lib/gracenote/auth/registration.rb

```diff
@@ -2,13 +2,12 @@ module Gracenote
   module Auth
     module Registration
       def register
-        query("register", client: client_id) do |response|
-          if response.ok?
-            @user_id = response.params["user"]
-          else
-            raise response.message
-          end
+        response = query("register") do |xml|
+          xml.CLIENT client_id
         end
+
+        raise Gracenote::Errors::RequestError.new(response.message) unless response.ok?
```

Comment:
__[rubocop]__
* Provide an exception class and message as arguments to `raise`. [:ref](https://github.com/bbatsov/ruby-style-guide#exception-class-messages)
* Use `fail` instead of `raise` to signal exceptions. [:ref](https://github.com/bbatsov/ruby-style-guide#fail-method)